### PR TITLE
fix: Include barcode format on iOS when using mlkit

### PR DESCRIPTION
**Pull request by joshea0**
# Pull Request

### Change Description

> Summary of the change. Please also include relevant motivation and context.

**Type of change:** (select all that apply)

- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Change requiring an ADR
- [ ] Other (please specify)
 
## Dependencies

> Dependencies that are required for this change.

N/A

### Testing Summary

> ​Description of tests run to verify changes, including instructions to reproduce .

- Build on iOS 
- Scan a barcode using MLkit and confirm the `format` field on the `Barcode` object is defined and has the correct value

### Testing Configuration

> Relevant details of your testing configuration

Tested on iPhone 8 with iOS 15.2.1

### Test Outcomes

> Tests listed below should correspond to the ones listed in your summary

- [ ] Scan a barcode using MLkit and confirm the `format` field on the `Barcode` object is defined and has the correct value

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code if needed, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if needed
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or verified that my feature works
- [ ] New and existing unit tests pass locally with my changes

Unchecked items above are because I am unsure what the current state of style guidelines and unit tests are in this project, since this is the first PR in our fork.

## Pull Request Review Template

> Information for the Pull Request Reviewer

- [x] I have read through the code and provided comments that follow - please use <https://conventionalcomments.org/>
- [ ] I have built the code locally and tested the linked tickets acceptence criteria (Ignore for small changes)
- [ ] I have build the code on App Center and Deployed (If necessary)

### CODE REVIEW

File: ios/RN/BarcodeDetectorManagerMlkit.m [hunk 1]
```diff
@@ -119,6 +120,8 @@ - (NSArray *)processBarcodes:(NSArray *)barcodes imageContainingBarcodes:(UIImag
 
         MLKBarcodeValueType valueType = barcode.valueType;
         [resultDict setObject:[self getType:barcode.valueType] forKey:@"type"];
+        NSString *getFormatResult = [self getFormat:barcode.format];
```
Comment (mljlynch) [inline ios/RN/BarcodeDetectorManagerMlkit.m]: question: is this just a header to set the type definition?

File: ios/RN/BarcodeDetectorManagerMlkit.m [hunk 2]
```diff
@@ -242,6 +245,54 @@ - (NSArray *)processBarcodes:(NSArray *)barcodes imageContainingBarcodes:(UIImag
     return result;
 }
 
+- (NSString *)getFormat:(long)type
+{
+    NSString *barcodeType = @"UNKNOWN";
```
Comment (mljlynch) [inline ios/RN/BarcodeDetectorManagerMlkit.m:250]: I don't really know about objective-c, but this seems like a pretty low risk change.

**Review by mljlynch — APPROVED** [review]

**Review by mljlynch — APPROVED** [review]
I don't really know about objective-c, but this seems like a pretty low risk change.

File: ios/RN/BarcodeDetectorManagerMlkit.m [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #import "BarcodeDetectorManagerMlkit.h"
 #import <React/RCTConvert.h>
+#import <React/RCTLog.h>
```
Comment (mljlynch) [inline ios/RN/BarcodeDetectorManagerMlkit.m]: question: why was this lib imported?

Comment (joshea0) [reply ios/RN/BarcodeDetectorManagerMlkit.m]: that was for logging. I should have removed it before pushing, my mistake. I'll fix it shortly.

File: ios/RN/BarcodeDetectorManagerMlkit.m (hunk 1 above)
Comment (joshea0) [reply ios/RN/BarcodeDetectorManagerMlkit.m]: I don't think I understand this question, can you please clarify? I don't know what you mean by "header" in this context.

This line is converting the `barcode.format` value (which is an enum) to `NSString` (just a string in objective-c). Then the next line sets that string value in the `resultDict` which is the object we eventually pass back as `barcode` to the callbacks in the JS side of the react-native app

**Pull request merged**